src/snowflake/connector/connection.py

```diff
@@ -349,6 +365,17 @@ def __init__(
                     f" known ones are {list(connections.keys())}"
                 )
             kwargs = {**connections[connection_name], **kwargs}
+        elif empty_kwargs:
+            # connection_name is None and kwargs was empty when called
+            def_connection_name = CONFIG_MANAGER["default_connection_name"]
+            connections = CONFIG_MANAGER["connections"]
```

Comment:
Similarly, if I happen to have a config file that has no "connections" section, this will raise a KeyError. A meaningful error is better.

Reply:
Okay, I tried this out manually. I found 2 bugs that made things not work how I thought they should.
When I remove my `connections.toml` file I now get:
```
Traceback (most recent call last):
  File "/tmp/autoconnect.py", line 3, in <module>
    with connect() as conn:
         ^^^^^^^^^
  File "/Users/mkeller/snowflakedb/snowflake-connector-python/src/snowflake/connector/__init__.py", line 54, in Connect
    return SnowflakeConnection(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mkeller/snowflakedb/snowflake-connector-python/src/snowflake/connector/connection.py", line 373, in __init__
    raise Error(
snowflake.connector.errors.Error: Default connection with name 'mkeller_sfctest0_aws' cannot be found, known ones are []
```
I also assigned a default value to `connections`, so hopefully that will cover all the use-cases and give better error messages.